MMM-Volume.js

```diff
@@ -40,18 +48,25 @@ Module.register("MMM-Volume", {
         setVolumeScript: `osascript -e 'set volume output volume #VOLUME#'` // set 0~100
       },
       "ALSA": {
-       getVolumeScript: `amixer sget 'PCM' | awk -F"[][]" '{print ""$2""}' | grep %  | awk '{gsub ( /%/, "" ) ; print}'`, // get 0~100
+        getVolumeScript: `amixer sget 'PCM' | awk -F"[][]" '{print ""$2""}' | grep %  | awk '{gsub ( /%/, "" ) ; print}'`, // get 0~100
         setVolumeScript: `amixer sset -M 'PCM' #VOLUME#%`, //set 0~100
       },
       "HIFIBERRY-DAC": {
         getVolumeScript: `amixer sget 'Digital' | grep -E -o '[[:digit:]]+%' | head -n 1| sed 's/%//g'`, // get 0~100
         setVolumeScript: `amixer sset -M 'Digital' #VOLUME#%`, // set 0~100
       },
- 	"PULSE": {
+ 	    "PULSE": {
         getVolumeScript: `amixer get Master  | awk -F"[][]" '{print ""$2""}' | grep %  | awk 'NR==1{print $1}' | awk '{gsub(/%/,"") ; print}'`, // get 0~100
         setVolumeScript: `amixer set Master #VOLUME#% -q`, // set 0~100
       },
-
+      "SPEAKER": {
```

Comment:
@xuepgao I have some questions. The device name "Speaker" and "Playback" are the default values for Respeaker 2-mic voice card? I think they might be the result of ALSA configuration by user(you), So I'm not sure these values are unique for the device.

And why `Speaker` and `Playback` are divided? Because it has 2 speakers by default? So you need to control them one by one? but my module is not designed to control multi-channels.

The last thing, I have no this device so I cannot test it by myself. Anyway, to get Volume values (0~100 integer), only `amixer sget DEVICENAME` enough? I think more additional shell commands (like grep, awk) might be needed.

Reply:
hi


I’m on business travel today . So it is not convenient to test the voice card. 

Sorry for PR, no description, no detail test... Frankly speaking, before met you I even don’t know what does PR mean. I promise you I submit a PR... You can make the PR pending until I do well everything.

The follow link are some logs when I used amixer command:
https://docs.google.com/document/d/1-S0OnwFiX8aDaENcO9qKEay6bQ-vn5GN8phVKY7kPGk/mobilebasic


The follow link are other people’s command reference:
https://github.com/respeaker/respeaker_openwrt_files/blob/master/files/usr/bin/volume

I have the same concern as you, because “Speaker” or “Playback” doesn’t seem like a unique device id. But the seeed SE told me it’s a real correct command. Maybe it is a reason because I set Respeaker 2mic as default voice card? Let me remove the voice card from RPI and test these commands. Anyway, I will continue to test it ASAP when I go back home.